### DISCUSSION TITLE
[question] run nerdctl as root but without sudo

### DISCUSSION THREAD

**Discussion by shyim** [containerd/nerdctl#766] [Q&A] [answered]
Hey,

is it possible to use nerdctl on normal account but with root mode? Like in docker adding the current user to the docker group 🤔

**Comment by tigerinus** [comment]
same question here - any anwser yet?

**Comment by AkihiroSuda** [comment]
> is it possible to use nerdctl on normal account but with root mode? Like in docker adding the current user to the docker group 

No, due to https://github.com/containerd/nerdctl/issues/473#issuecomment-953548961 .

Workaround is `alias nerdctl=sudo nerdctl`, or set SETUID on `nerdctl` binary.
(In the latter case, the nerdctl binary should be placed in `~/bin`, and chmod 700 `~/bin`)

**Comment by AkihiroSuda** [comment, marked as answer]
Added a documentation about this in https://github.com/containerd/nerdctl/blob/master/docs/faq.md#does-nerdctl-have-an-equivalent-of-sudo-usermod--ag-docker-user-

```
mkdir -p $HOME/bin
chmod 700 $HOME/bin
cp /usr/local/bin/nerdctl $HOME/bin
sudo chown root $HOME/bin/nerdctl
sudo chmod +s $HOME/bin/nerdctl
export PATH=$HOME/bin:$PATH
```

**Reply by jshk00** [reply]
Well this will not works because when you try to stop container it throws the iptables error which requieres sudo to run.

**Reply by AkihiroSuda** [reply]
What error?

**Reply by jshk00** [reply]
ERRO[0000] failed to call cni.Remove                     error="plugin type=\"firewall\" failed (delete): could not initialize iptables protocol 0: could not get iptables version: exit status 111"

apparently you cannot stop container with this setup

**Reply by AkihiroSuda** [reply]
Does the error happen with `sudo nerdctl` too? If so, the issue is irrelevant to `chmod +s`

**Reply by jshk00** [reply]
no it does not happens with `sudo nerdctl` only when i use just `nerdctl`

**Reply by AkihiroSuda** [reply]
Thanks for reporting, opened an issue:
- https://github.com/containerd/nerdctl/issues/4098

**Reply by AkihiroSuda** [reply]
A workaround would be to just set `alias nerdctl=sudo nerdctl`.

This does not have any negative impact on security, as `sudo chmod +s $HOME/bin/nerdctl` is already insecure.

It is highly suggested to use Rootless mode instead: https://rootlesscontaine.rs/getting-started/containerd/

**Comment by arnoldas500** [comment]
Is there a solution to this yet? Having to do the above for every k8s user on the system doesnt seem like a good option if you have many users.

**Reply by AkihiroSuda** [reply]
You can place the setuid-enabled `nerdctl` binary into a directory that is accessed by a specific GID (`chmod 750`), but you shouldn't give the root to that "many users"?

**Reply by arnoldas500** [reply]
Thanks for sharing that. 

Is there possibly a better tool than nerdctl to allow users to commit running containers managed by k8s? I switch from docker to containerd and nerdctl but it also doesnt seem like a good option.
